numpy/lib/function_base.py

```diff
@@ -1801,6 +1801,10 @@ def place(arr, mask, vals):
            [44, 55, 44]])
 
     """
+    if not isinstance(arr, np.ndarray):
```

Comment:
What error message do you get when `place` is not valid? What sort of duck typing does it currently support, if any?

It might be worth updating `_insert` instead.